Cython/Includes/cpython/longintrepr.pxd

```diff
@@ -2,14 +2,14 @@
 # This is not part of Python's published API.
 
 cdef extern from "longintrepr.h":
-    # Add explicit cast to avoid compiler warnings
-    cdef _PyLong_New "(PyObject*)_PyLong_New"(Py_ssize_t s)
-
     ctypedef unsigned int digit
     ctypedef int sdigit  # Python >= 2.7 only
 
-    ctypedef struct PyLongObject:
-        digit* ob_digit
+    ctypedef class __builtin__.longint [object PyLongObject]:
+        cdef digit* ob_digit
+
+    # Add explicit (PyObject *) cast to avoid compiler warnings
+    cdef longint _PyLong_New "(PyObject *)_PyLong_New"(Py_ssize_t s)
```

Comment:
Should we be casting to PyLongObject* instead of PyObject* here?

Reply:
> Should we be casting to PyLongObject* instead of PyObject* here?

No, the `PyObject *` really is intentional because `_PyLong_New` returns a `PyLongObject *`. I'm not entirely sure whether that cast is still needed, it seems that Cython already does a cast to `PyObject *` anyway:
```
__pyx_t_2 = ((PyObject *)(PyObject *)_PyLong_New(__pyx_t_1)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 4, __pyx_L1_error)
```